app/helpers/layout_helper.rb

```diff
@@ -347,6 +347,20 @@ def fullscreen_button(element = "$(this).prev()")
   private
 
   def authorized_associations(associations)
-    associations.included_modules.include?(Authorizable) ? associations.authorized : associations
+    if associations.included_modules.include?(Authorizable)
+      if associations.respond_to?(:klass)
+        associations.authorized(authorized_assocations_permission_name(associations.klass), associations.klass)
+      else
+        associations.authorized(authorized_assocations_permission_name(associations), associations)
+      end
+    else
+      associations
+    end
+  end
+
+  def authorized_assocations_permission_name(klass)
+    permission = "view_#{klass.to_s.underscore.pluralize}"
+    return nil unless Permission.where(:name => permission).present?
```

Comment:
I think the whole method (return permission or nil) is just the equivalent of 
```ruby
Permission.find_by_name("view_#{klass.to_s.underscore.pluralize}").try(:name)
```


Since `find_by_name` will not work on Rails 4, just leaving a note `# TODO: Substitute by find_by :name for Rails 4` should be enough. To me the one-liner looks more understandable/maintainable, change only if it makes sense to you too.